### PR TITLE
fluidsynth: update to 2.2.3

### PR DESCRIPTION
**Pull request by kcgen**

### CODE REVIEW

**Comment by kcgen** [comment]
Fluidsynth is listed under `skip_windows` in the CI exclusions, however it fails the visual studio CI check (which is expected). 

Is there something else needed to exclude/pass the visual studio CI check?

**Comment by xclaesse** [comment]
@kcgen skip_windows means it's known to fail but the next one who update that package is in charge of fixing it. Congratulation you got selected :D

File: ci_config.json [hunk 1]
```diff
@@ -21,9 +21,13 @@
     "sdl2_net", "sdl2_ttf", "termbox", "unit-system", "xxhash", "zstd"
   ],
 
+  "fluidsynth": {
+    "linux_only": true
```
Comment (xclaesse) [inline ci_config.json:25]: linux_only should only be used for projects that cannot work on other platforms, for example when they are made for a specific Linux kernel module. Is it the case here?

**Comment by kcgen** [comment]
> IIRC the error I saw in previous CI run, the problem was in proxy-libintl and is already fixed in [frida/proxy-libintl@50bd252](https://github.com/frida/proxy-libintl/commit/50bd2525261d44d80533a523873b9344a6d741c5).

Right - it's bombing out on Windows in the third layer of dependencies: `Executing subproject fluidsynth:glib:proxy-libintl`

```text
Executing subproject fluidsynth:glib:zlib

zlib| Project name: zlib
zlib| Project version: 1.2.11
zlib| C compiler for the host machine: cl (msvc 19.29.30136 "Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30136 for x64")
zlib| C linker for the host machine: link link 14.29.30136.0
zlib| Windows resource compiler: Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
zlib| Build targets in project: 5
zlib| Subproject zlib finished.

glib| Checking for function "ngettext" : NO
glib| Library intl found: NO
glib| Using subprojects\glib-2.68.1\subprojects\proxy-libintl.wrap

Executing subproject fluidsynth:glib:proxy-libintl

proxy-libintl| Project name: proxy-libintl
proxy-libintl| Project version: 1
proxy-libintl| C compiler for the host machine: cl (msvc 19.29.30136 "Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30136 for x64")
proxy-libintl| C linker for the host machine: link link 14.29.30136.0
proxy-libintl| WARNING: extract_all_objects called without setting recursive
proxy-libintl| keyword argument. Meson currently defaults to
proxy-libintl| non-recursive to maintain backward compatibility but
proxy-libintl| the default will be changed in the future.


subprojects\proxy-libintl\meson.build:15:0: ERROR: Fatal warnings enabled, aborting

A full log can be found at D:\a\wrapdb\wrapdb\_build\meson-logs\meson-log.txt
.
======================================================================
ERROR: test_releases (__main__.TestReleases) (step='check_new_release', name='fluidsynth')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tools/sanity_checks.py", line 169, in test_releases
    self.check_new_release(name)
  File "tools/sanity_checks.py", line 240, in check_new_release
    subprocess.check_call(['meson', 'setup', builddir] + options)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['meson', 'setup', '_build', '--fatal-meson-warnings', '-Dwraps=fluidsynth', '-Dpython.platlibdir=C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\Lib\\site-packages', '-Dpython.purelibdir=C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\Lib\\site-packages']' returned non-zero exit status 1.

----------------------------------------------------------------------
Ran 3 tests in 62.716s

FAILED (errors=1, skipped=1)
Error: Process completed with exit code 1.
```

If it's taken care of in  frida/proxy-libintl@50bd252, what does that mean for us here - will this fix be absorbed here, then I can rebase?

**Comment by eli-schwartz** [comment]
I previously submitted multiple MRs on the gnome gitlab to fix warnings in the build. Now I've gone back and submitted another MR to backport those fixes into the latest stable release, after which I hope to update the glib wrap to 2.70.1 and everything should work...

**Comment by kcgen** [comment]
With https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2304 merged, are we still waiting on other fixes or PRs to propagate?

(to sanity check, I reconfirmed my master in-sync)
```
[09:48:22@server wrapdb] (master) git fetch upstream
[09:48:28@server wrapdb] (master) git merge upstream/master
Already up to date.
```

**Comment by eli-schwartz** [comment]
The next glib release is due in 4 days...

**Pull request closed without merging**